### ISSUE TITLE
prepend_dirs (-d, -dd) not honored by several modules

### ISSUE THREAD

**Issue by epruesse** [ewels/MultiQC#365]
The `-d` and `-dd` flags are not honored by several modules. The corresponding configuration variable `config.prepend_dirs` is only used in `BaseModule.clean_s_name()`, which is not applied to the sample names by these modules:

 - bamtools
 - bismark
 - fastq_screen
 - gatk 
 - methylQA
 - preseq
 - rseqc
 - quast
 - samtools
 - snpeff
 - star

The prepend_dirs feature is very useful for generating reports to illustrate the effects of using differing tools or parameters. In those cases, the sample name will often be identical but the path will vary.

The name of the `clean_s_name` function is somewhat misleading, possibly warranting a rename to make it more obvious to module writers that `clean_s_name` must be applied to the sample names at some point.

**Comment by ewels** [comment]
Many thanks @epruesse, as you say - the current function name probably isn't ideal and is caused by the way that this code has grown historically. At first the function basically just chopped off file extensions, but it's slowly grown to encompass more and more functionality, `-d` being one of the more important features now.

I was a bit shocked when I saw the length of your list above - I guess that you've got this by looking at which module files contain reference to the `clean_s_name` function. Thankfully, most of the modules you list don't need to. This is because they all get their sample names from the log filename, through the `f['s_name']` variable. This name has already had `clean_s_name` applied to it in the base module before being returned. The only time that this function needs to be applied within the module code is when sample names are gathered from the file contents instead of file name, as with the QUAST module that you fixed.

The exception to this is methylQA, which optionally overwrites the `f['s_name']` name with a custom one. I've added the `clean_s_name` function in here.

I hope this makes sense! This is loosely describe in [the docs](http://multiqc.info/docs/#custom-sample-names) but it's not as clear as it could be, I'll add a note emphasising that the function must be applied for custom sample names so that it's more obvious.

Phil
